src/app/base/components/DhcpFormFields/DhcpFormFields.tsx

```diff
@@ -134,21 +131,30 @@ export const DhcpFormFields = ({ editing }: Props): JSX.Element => {
           { value: "device", label: "Device" },
         ]}
       />
-      {type &&
+      {type === "machine" ? (
+        <MachineSelect
```

Comment:
This will be followed up in https://github.com/canonical/maas-ui/issues/4397

Reply:
I think this is worth doing in this PR as it doesn't relate to `ContextualMenu`.

Reply:
I am happy to do that, but would you mind if I did this separately in https://github.com/canonical/app-tribe/issues/1340.

I prefer to move in small increments in each PR, building on top of changes from previous work. This can be very helpful especially in case of us working in different time zones and the lag in code reviews.